### PR TITLE
ci: set up the basic github workflow

### PR DESCRIPTION
**Pull request by yanjing1104**

### CODE REVIEW

**Comment by yanjing1104** [comment]
Comment "Lint check", "Shell check", "Golangci lint check" and "Cyclomatic check" in the current PR, cause the current codebase will fail these checks in the github workflow.
Will push more PRs to fix the failures, and then uncomment them.
Thus mark this PR as WIP now.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        make bin/memtierd
+        make bin/meme
+
+    - name: Unit test
+      run: make race=1 test
+
+    - name: Fuzz test
+      run: make race=1 fuzztest
+
+    - name: Install dependencies
```
Comment (askervin) [inline .github/workflows/main.yml:33]: Nit: slightly more descriptive name would be better: install dependencies for what? ("static checks"?)

**Comment by yanjing1104** [comment]
> Cool, this is great, Jing!
> 
> I'm not merging this yet, as it is marked as a Draft PR, but will be more than happy to take it in whenever you think it's good enough to start with.

Hi Antti I think it might take quite some time to fix the "Shell check", "Golangci lint check" and "Cyclomatic check", maybe it would be better to get this workflow to work first and uncomment the related code later. 

I updated this PR as ready for review, pls take a look at it, thank you. :D

**Comment by yanjing1104** [comment]
Closed this PR as the code has been merged to the main branch.

**Pull request closed without merging**